packages/typescript-resolver-files/src/parseSources/parseSources.ts

```diff
@@ -21,7 +24,10 @@ export function parseSources(sources: Source[]): ParseSourcesResult {
       const sourcePath = path.parse(source.location);
       const moduleDir = sourcePath.dir;
 
-      const moduleName = path.basename(moduleDir);
+      const moduleName = path.relative(
+        path.resolve(baseOutputDir),
+        path.basename(moduleDir)
+      );
```

Comment:
`moduleName` is eventually used to identify module files belong to, so if we use this as filepath, it'd result in errors like [this](https://github.com/eddeee888/graphql-code-generator-plugins/actions/runs/8446147770/job/23275844474?pr=244#step:5:731).

In this case, I think it's better to create variable like "relative path from `baseOutputDir` to module dir" and use it instead of [moduleName here](https://github.com/eddeee888/graphql-code-generator-plugins/blob/9d9657fa54d1bb9c1c4b8ca0cf75a3ac3aaba754/packages/typescript-resolver-files/src/generateResolverFiles/visitNamedType.ts#L159).